### PR TITLE
tf.broadcastTo

### PR DESCRIPTION
**Pull request by DirkToewe**
Implements `tf.broadcastTo` as requested in #2011.The implementation uses the backend implementation of `tf.tile` and should therefore be just as fast.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2238)

### CODE REVIEW

File: tfjs-core/src/ops/array_ops.ts [hunk 1]
```diff
@@ -26,6 +26,72 @@ import {op} from './operation';
 import {MPRandGauss, RandGamma, UniformRandom} from './rand';
 import {zeros, zerosLike} from './tensor_ops';
 
+/** Broadcast an array to a compatible shape NumPy-style.
+ *  
+ *  The tensor's shape is compared to the broadcast shape from end to beginning.
+ *  Ones are prepended to the tensor's shape until is has the same length as
+ *  the broadcast shape. If input.shape[i]==shape[i], they (i+1)-th axis is
```
Comment (annxingyuan) [inline tfjs-core/src/ops/array_ops.ts]: nit: the (i+1)-th

File: tfjs-core/src/ops/array_ops.ts [hunk 2]
```diff
@@ -26,6 +26,72 @@ import {op} from './operation';
...
+  {
+    const newShape =  input.shape.slice();
+    while(newShape.length < shape.length ) {
+          newShape.unshift(1);
+    }
+    input = input.reshape(newShape);
+  }
+
+  const reps: number[] = Array.from(shape);
+  for( let i=shape.length; i-- > 0; )
```
Comment (annxingyuan) [inline tfjs-core/src/ops/array_ops.ts]: let i=shape.length; i>0; i--

File: tfjs-core/src/ops/array_ops.ts [hunk 3]
```diff
@@ -26,6 +26,72 @@ import {op} from './operation';
...
+      throw new Error(
+        `broadcastTo(): [${xShape}] cannot be broadcast to [${shape}].`
+      );
+    }
+  }
+
+  const axes = reps.map( ( n,i)  => n > 1 ? i : -1 ).filter( i => i >= 0 );
+
+  if( axes.length === 0 ) {
+    return input as Tensor<R>;
```
Comment (annxingyuan) [inline tfjs-core/src/ops/array_ops.ts]: Please clone the result before returning it, like this: `input.clone()`.

Comment (DirkToewe) [reply tfjs-core/src/ops/array_ops.ts]: Without `.clone()`, disposing the input would also dispose the return value, right? Didn't even think about that.

File: tfjs-core/src/ops/array_ops.ts (hunk 1 above)
Comment (DirkToewe) [reply tfjs-core/src/ops/array_ops.ts]: Changed it.

File: tfjs-core/src/ops/array_ops.ts (hunk 2 above)
Comment (DirkToewe) [reply tfjs-core/src/ops/array_ops.ts]: Changed.

**Comment by annxingyuan** [comment]
@DirkToewe sorry just one last thing - could you use `expectArraysClose` everywhere, instead of checking for mobile safari? This is what we do in our other tests. After that we'll be good to go :)

**Comment by DirkToewe** [comment]
Okay, I changed it. In [tf.linalg.bandPart](https://github.com/tensorflow/tfjs/pull/2226/files#diff-7eb4288fe3d7c2d1522e5a1bce92e087R134-R136), I've used the same check. If that's okay I'm going to changed it in that commit as well.

**Comment by annxingyuan** [comment]
@DirkToewe that would be great! thanks!

**Pull request merged**